back/boxtribute_server/graph_ql/filtering.py

```diff
@@ -73,9 +79,34 @@ def derive_box_filter(filter_input):
     product_gender = filter_input.get("product_gender")
     if product_gender is not None:
         condition &= Box.product.gender == product_gender
+        join_with_product_required = True
 
     product_category_id = filter_input.get("product_category_id")
     if product_category_id is not None:
         condition &= Box.product.category == product_category_id
+        join_with_product_required = True
+
+    product_id = filter_input.get("product_id")
+    if product_id is not None:
+        condition &= Box.product == product_id
+
+    size_id = filter_input.get("size_id")
+    if size_id is not None:
+        condition &= Box.size == size_id
+
+    tag_ids = filter_input.get("tag_ids")
+    if tag_ids is not None:
+        selection = selection.join(
+            TagsRelation,
+            src=Box,
+            on=(
+                (TagsRelation.object_type == TaggableObjectType.Box)
+                & (TagsRelation.object_id == Box.id)
+                & (TagsRelation.tag << tag_ids)
+            ),
+        )
```

Comment:
todo: check if this needs distinct

Reply:
Yes, you will need to change something here. This query returns duplicate results:
```
{boxes(
  baseId:1, 
  filterInput:{
    tagIds: [4,5]
    
  }) {
    elements {id}}}
    ```

Reply:
There are two ways to solve this. See case 4.) in this [link](https://learnsql.com/blog/get-rid-of-duplicates-sql-join/)